### ISSUE TITLE
Support for Timezone setting for the dev profile

### ISSUE THREAD

**Issue by mcansky** [joemasilotti/railsdevs.com#49]
As a developer based in EU I want to be able to select my TZ so that potential clients can figure out if I match their timezone enough when looking at my profile.

## UI bits

- when editing a developer profile, display a select box with TZ to select from using City, Country "Paris, France" and UTC reference "UTC +/- 1".
- when showing a developer profile, display the timezone.

## Questions

- is it worth digging into the whole "where are you based?" with country too ?

**Comment by joemasilotti** [comment]
I love this idea! How are you planning on mapping city/country to a timezone? I think you might be able to map `ActiveSupport::TimeZone.all` but that might be a bit expensive. (It takes a split second to load in the console, at least.) Would love to hear your thoughts – ideally we don't need another gem for this unless absolutely necessary.

```ruby
[18] pry(main)> ActiveSupport::TimeZone.all.map { |tz| [tz.tzinfo.name, tz.name] }.to_h
=> {"Etc/GMT+12"=>"International Date Line West",
 "Pacific/Pago_Pago"=>"American Samoa",
 "Pacific/Midway"=>"Midway Island",
 "Pacific/Honolulu"=>"Hawaii",
 "America/Juneau"=>"Alaska",
 "America/Los_Angeles"=>"Pacific Time (US & Canada)",
 "America/Tijuana"=>"Tijuana",
 "America/Phoenix"=>"Arizona",
 "America/Chihuahua"=>"Chihuahua",
 "America/Mazatlan"=>"Mazatlan",
 "America/Denver"=>"Mountain Time (US & Canada)",
...
```

> is it worth digging into the whole "where are you based?" with country too ?

Maybe, but I think that can be tracked as a separate issue. For now, let's only handle timezones.

**Comment by joemasilotti** [comment]
Oh, I guess [`time_zone_select`](https://apidock.com/rails/v5.2.3/ActionView/Helpers/FormOptionsHelper/time_zone_select) should work out nicely after some TailwindUI styling! Unless you had something else in mind.

<img width="424" alt="image" src="https://user-images.githubusercontent.com/2092156/140611776-c789f329-79da-4777-ae3c-4582b53f9c54.png">

**Comment by mcansky** [comment]
to be honest I did not think about the how the solution would work. I mostly wanted to point out the need for something to help companies find someone in a "compatible" time zone, start the discussion and see where we will go from there.

`time_zone_select` looks about what I had in mind for the look and quite similar to what we are used to from other sites.

**Comment by caseyjenks** [comment]
@joemasilotti Is there a need to support filtering developers by timezone? Or are we just wanting to display this on the profiles for now.

**Comment by joemasilotti** [comment]
I think for now all we need to do is display it on the profile.

That said, I'd love to see design mockups or ideas of _where_ this will go before someone dives into implementation. I'm worried the developer profile page is getting a little busy.

**Comment by caseyjenks** [comment]
@joemasilotti Thoughts on putting it here? Seems like the best spot, unless you think it could live under the loosely related availability section. 
![image](https://user-images.githubusercontent.com/300131/144659770-e6ee7c99-af61-4d67-949c-749b21b75b16.png)

**Comment by joemasilotti** [comment]
Yeah, this work! Good idea, and I love the clock.

**Comment by joemasilotti** [comment]
Closed via #138.
